src/Symfony/Bridge/Twig/Tests/Extension/HttpKernelExtensionTest.php

```diff
@@ -70,11 +70,17 @@ public function testGenerateFragmentUri()
 
         $kernelRuntime = new HttpKernelRuntime($fragmentHandler, $fragmentUriGenerator);
 
+        $controller = TemplateController::class;
+
+        if (Environment::VERSION_ID >= 31200) {
+            $controller = str_replace('\\', '', TemplateController::class);
```

Comment:
I don't understand why this is needed. Do you have a link to the failing test in the TwigBridge?

Reply:
see https://github.com/symfony/symfony/actions/runs/10341620300/job/28623626714?pr=57968#step:7:2696 for example

Reply:
and if you take a look at the URL that we compare on line 93, you can also see that Twig internally handles the class name without backslashes

Reply:
So the solution would be to escape the `\\` so that it becomes:
```
Symfony\\Bundle\\FrameworkBundle\\Controller\\TemplateController
```

Reply:
```suggestion
            $controller = str_replace('\\', '\\\\', TemplateController::class);
```

Reply:
So that means this test was always flawed already?

I think the EnvironmentId check should be removed. And always do the double escape + fix the assertion.

Reply:
I did that first, but then reverted back to this solution as I felt it was wrong to also have to change the assertion. But I guess you are right that the test was not correct before and having to fix the assertion is indeed the way to go.